src/core/server/saved_objects/migrations/model/extract_errors.ts

```diff
@@ -51,3 +51,32 @@ export function extractUnknownDocFailureReason(
     `'`
   );
 }
+
+/**
+ * Constructs migration failure message string for doc exceeds max batch size in bytes
+ */
+export const fatalReasonDocumentExceedsMaxBatchSizeBytes = ({
+  _id,
+  docSizeBytes,
+  maxBatchSizeBytes,
+}: {
+  _id: string;
+  docSizeBytes: number;
+  maxBatchSizeBytes: number;
+}) =>
+  `The document with _id "${_id}" is ${docSizeBytes} bytes which exceeds the configured maximum batch size of ${maxBatchSizeBytes} bytes. To proceed, please increase the 'migrations.maxBatchSizeBytes' Kibana configuration option and ensure that the Elasticsearch 'http.max_content_length' configuration option is set to an equal or larger value.`;
+
+/**
+ * Constructs migration failure message and logs message strings when an unsupported cluster routing allocation is configured.
+ * The full errorMessage is "[unsupported_cluster_routing_allocation] The elasticsearch cluster has cluster routing allocation incorrectly set for migrations to continue."
```

Comment:
More for information and code search than anything else

Reply:
nit: TBH it might make things more readable to just concat the messages in `model.ts` instead of doing it inside the function. e.g.

```js
reason: `${left.message} ${formatClusterRoutingAllocationFailureReason(docLink)}`,
```

Reply:
we might decide to change the `doclink` section we're pointing to or extend the error message later on. Also, given that we've got other error message generation functions, I'd like to keep the helper function as it is.